File: 121.Best Time to Buy and Sell Stock/solution.py
def maxProfit(prices):
    m = 0
    stk = [prices[0]]
    for i in range(1,len(prices)):
        if stk[-1] > prices[i]:
            stk.pop()
            stk.append(prices[i])
        else:
            m = max(m, prices[i] - stk[-1])
    return m

print(maxProfit([7,1,5,3,6,4]))

File: 121.Best Time to Buy and Sell Stock/Solution.cpp
#include <bits/stdc++.h>

using namespace std;


int maxProfit(vector<int>& prices) {
    stack<int> stk;
    stk.push(prices[0]);
    int max_profit = 0;


    for (int i = 1; i < prices.size(); i++){
        if (stk.top() > prices[i]){
            stk.pop();
            stk.push(prices[i]);
        }else{
            max_profit = max(max_profit, prices[i] - stk.top());
        }
    }

    return max_profit;


}